fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/clientlog/KeySpaceCountTree.java

```diff
@@ -548,24 +573,23 @@ protected boolean isRecordStoreLeaf(@Nonnull FDBRecordContext context, @Nonnull
     }
 
     protected CompletableFuture<Resolved> resolvePrimaryKeyField(@Nonnull FDBRecordContext context, @Nonnull Resolved resolvedParent, @Nullable Object object,
-                                                                 @Nonnull KeyExpression primaryKey, int index) {
-        List<KeyExpression> keys = primaryKey.normalizeKeyForPositions();
-        if (index < keys.size()) {
-            KeyExpression key = keys.get(index);
-            while (key instanceof NestingKeyExpression) {
-                key = ((NestingKeyExpression)key).getChild();
-            }
-            if (key instanceof FieldKeyExpression) {
-                return CompletableFuture.completedFuture(new ResolvedPrimaryKeyField(resolvedParent, ((FieldKeyExpression)key).getFieldName(), object, object));
-            }
-        }
-        return UNRESOLVED;
+                                                                 @Nonnull KeyExpression fieldKey, int fieldIndex) {
```

Comment:
It looks like `fieldIndex` can be removed as a parameter from both this and `resolveIndexField`.

Reply:
I was thinking an overriding method might understand that the zeroth field of the primary key is special, for example. But there are probably other ways it could accomplish such a check.

Reply:
Okay, I see. Well, it probably doesn't hurt to leave it like that. I'll approve, and if you want to leave it as is, then I guess feel free to merge.